app/src/main/java/com/example/makeiteven2/adapters/LevelsAdapter.kt

```diff
@@ -10,15 +10,31 @@ import androidx.recyclerview.widget.RecyclerView
 import com.example.makeiteven2.R
 import com.example.makeiteven2.data_models.Level
 import com.example.makeiteven2.extras.OnTouchAnimation
+import com.example.makeiteven2.fragments.FragmentLevelsScreen
 import kotlinx.android.synthetic.main.level_cell.view.*
 
 class LevelsAdapter(private var mLevelsList : List<Level> = ArrayList(), private val context: Context,private var mCurrentStage : Int)
     : RecyclerView.Adapter<LevelsAdapter.ViewHolder>(){
 
+    var mCallBack : ILevelsAdapter
+
+    init {
+        if (context is ILevelsAdapter) {
+            mCallBack = context
+        }
+        else{
+            throw ClassCastException(context.toString() + "must implement ILevelAdapter")
+        }
+    }
+
+    interface ILevelsAdapter{
+        fun levelsAdapterItemClicked(levelNumber : Int)
+    }
+
     class ViewHolder(private val view : View,private val context: Context) : RecyclerView.ViewHolder(view){
 
         @SuppressLint("ClickableViewAccessibility")
```

Comment:
what does it mean?

Reply:
what do you mean?

Reply:
@SuppressLint("ClickableViewAccessibility") ??

Reply:
oo it is to suppress warnings